tests/function-controller/pkg/resource/resource.go

```diff
@@ -17,20 +16,24 @@ import (
 	watchtools "k8s.io/client-go/tools/watch"
 )
 
+type logger interface {
+	Logf(format string, args ...interface{})
+}
+
```

Comment:
you've already defined this interface in apirule.go - move this to some kind of shared place